### PR TITLE
add noScriptInfo checkboxes to allow scripts by origin

### LINKED ISSUE #6431: Feature to allow scripts by origin
The site shield does a good job in blocking all the scripts that are executed on the page but I would like to allow specific scripts to be executed on the page. On certain websites the video player scripts are blocked and there is no way to get the player working without disabling "script block" and "ad block".

I would like to suggest a feature to whitelist and blacklist certain scripts. Or feature similar to Ghostery where I could allow specific domain scripts to be executed.

#3732 #2383 #231

### PR DESCRIPTION
**Pull request by diracdeltas**
test plan:
1. disable scripts globally in about:preferences
2. go to vox.com. images should not appear.
3. click on the noscript icon
4. unselect everything except vox.com and vox-cdn.com and click one of the 'allow' buttons
5. images should appear
6. click on the noscript icon. you should see a bunch of origins but not vox.com and vox-cdn.com

fix #6431. adjusts the noScriptInfo dialog so that it can be used to allow scripts by origin, instead of all scripts on the page, when NoScript is globally enabled. removes the option to allow scripts persistently (since this can be done via the Bravery panel anyway)

auditors: @bbondy

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

screen shot 2017-02-15 at 10 45 06 pm

### CODE REVIEW

**Comment by bbondy** [comment]
ack sorry I had review comments pending and didn't notice.

File: less/forms.less [hunk 1]
```diff
@@ -494,14 +494,33 @@ select {
...
-    text-align: center;
+    max-width: 400px;
     font-size: 15px;
     cursor: default;
+    text-align: center;
 
     .truncate {
       margin-bottom: 5px;
     }
+    .noScriptCheckbox {
```
Comment (diracdeltas) [inline less/forms.less]: sgtm!

**Comment by diracdeltas** [comment]
@bbondy addressed review comments

**Comment by bbondy** [comment]
++

**Comment by k-ninja** [comment]
Why is there no option to permanently whitelist domains for script execution? It's not a great user experience to have to re-enable scripts for my internet banking every time I restart my browser.

**Pull request merged**